### PR TITLE
KOTS Upload in the Go API

### PR DESCRIPTION
**Pull request by marccampbell**
Implement the PUT to upload a new version (commonly linked to kots upload command) in the Go API.

The TS API continues to be present. 

This change was needed so that the preflight checks can run in-process instead of using the deploysocket and operator to execute.

### CODE REVIEW

File: api/src/controllers/kots/KotsAPI.ts [hunk 1]
```diff
@@ -227,6 +227,8 @@ export class KotsAPI {
     };
   }
 
+  // This handler was moved to kotsadm (go) in 1.14.0
+  // Remind me to delete this in 1.16.0
```
Comment (laverya) [inline api/src/controllers/kots/KotsAPI.ts]: why wait for 1.16 to delete it?

File: go.mod [hunk 2]
```diff
@@ -3,28 +3,18 @@ module github.com/replicatedhq/kotsadm
...
-	github.com/klauspost/cpuid v1.2.1 // indirect
-	github.com/konsorten/go-windows-terminal-sequences v1.0.2 // indirect
 	github.com/kubernetes-sigs/application v0.8.1 // indirect
 	github.com/lib/pq v1.3.0
 	github.com/mholt/archiver v3.1.1+incompatible
-	github.com/nwaples/rardecode v1.1.0 // indirect
-	github.com/pierrec/lz4 v2.4.1+incompatible // indirect
 	github.com/pkg/errors v0.9.1
 	github.com/replicatedhq/kots v1.13.10-0.20200317153303-89d5bfaddd77
+	github.com/replicatedhq/kotsadm/operator v0.0.0-20200317214647-13a394616e7f // indirect
```
Comment (laverya) [inline go.mod]: why is this required?

File: api/src/controllers/kots/KotsAPI.ts (hunk 1 above)
Comment (marccampbell) [reply api/src/controllers/kots/KotsAPI.ts]: I think there are some customers who have automation/integration tests that use this API directly, and it will give us a little time to help them move

File: go.mod (hunk 2 above)
Comment (marccampbell) [reply go.mod]: no idea, i'll tidy
